server/dex/dex.go

```diff
@@ -0,0 +1,254 @@
+		}
+
+		if symbol == "dcr" {
+			var ok bool
+			dcrBackend, ok = be.(*dcrasset.DCRBackend)
+			if !ok {
+				cancel()
+				return nil, fmt.Errorf("dcr backend is invalid")
+			}
+		}
+
+		ba := &asset.BackedAsset{
+			Asset: dex.Asset{
+				ID:       ID,
+				Symbol:   symbol,
+				LotSize:  assetConf.LotSize,
+				RateStep: assetConf.RateStep,
+				FeeRate:  assetConf.FeeRate,
+				SwapConf: assetConf.SwapConf,
+				FundConf: assetConf.FundConf,
+			},
+			Backend: be,
+		}
+
+		backedAssets[ID] = ba
+		lockableAssets[ID] = &swap.LockableAsset{
+			BackedAsset: ba,
+			CoinLocker:  dexCoinLocker.AssetLocker(ID).Swap(),
+		}
+	}
+
+	// Create DEXArchivist with the pg DB driver.
+	pgCfg := &pg.Config{
+		Host:          cfg.DBConf.Host,
+		Port:          strconv.Itoa(int(cfg.DBConf.Port)),
+		User:          cfg.DBConf.User,
+		Pass:          cfg.DBConf.Pass,
+		DBName:        cfg.DBConf.DBName,
+		HidePGConfig:  false,
+		QueryTimeout:  20 * time.Minute,
+		MarketCfg:     cfg.Markets,
+		CheckedStores: true,
+		Net:           cfg.Network,
+		FeeKey:        cfg.FeePubKey,
+	}
+	storage, err := db.Open(ctxDex, pgCfg)
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("db.Open: %v", err)
+	}
+
+	authCfg := auth.Config{
+		Storage: storage,
+		Signer:  cfg.DEXPrivKey,
+		//StartEpoch // ! different for each asset, and ambiguous
+		RegistrationFee: cfg.RegFeeAmount,
+		FeeConfs:        cfg.RegFeeConfirms,
+		FeeChecker:      dcrBackend.UnspentCoinDetails,
+	}
+
+	authMgr := auth.NewAuthManager(&authCfg)
+
+	// Create the swapper.
+	swapperCfg := &swap.Config{
+		Ctx:              ctxDex,
+		Assets:           lockableAssets,
+		Storage:          storage,
+		AuthManager:      authMgr,
+		BroadcastTimeout: 10 * time.Second,
+	}
+
+	swapper := swap.NewSwapper(swapperCfg)
+
+	// Markets
+	markets := make(map[string]*market.Market, len(cfg.Markets))
+	for _, mktInf := range cfg.Markets {
+		baseCoinLocker := dexCoinLocker.AssetLocker(mktInf.Base).Book()
+		quoteCoinLocker := dexCoinLocker.AssetLocker(mktInf.Quote).Book()
+		mkt, err := market.NewMarket(ctxDex, mktInf, storage, swapper, authMgr, baseCoinLocker, quoteCoinLocker)
+		if err != nil {
+			cancel()
+			return nil, fmt.Errorf("NewMarket failed: %v", err)
+		}
+		markets[mktInf.Name] = mkt
+	}
+
+	// Book router
+	bookSources := make(map[string]market.BookSource, len(cfg.Markets))
+	for mktName, mkt := range markets {
+		bookSources[mktName] = mkt
+	}
+	bookRouter := market.NewBookRouter(&market.BookRouterConfig{
+		Ctx:     ctxDex,
+		Sources: bookSources,
+		//EpochDuration: 10_000, // !!!!! different for each market
+	})
+
+	// Order router
+	marketTunnels := make(map[string]market.MarketTunnel, len(cfg.Markets))
+	for mktName, mkt := range markets {
+		marketTunnels[mktName] = mkt
+	}
+
+	orderRouter := market.NewOrderRouter(&market.OrderRouterConfig{
+		Assets:          backedAssets,
+		AuthManager:     authMgr,
+		MarketBuyBuffer: 0.1, // TODO: config
```

Comment:
Could be added to MarketTunnel so the order router can get each markets buy buffer.

Reply:
Now in MarketTunnel. https://github.com/decred/dcrdex/pull/81/commits/bce4b11a7ad079fa26d0644ae98782b189b39508